### PR TITLE
[chore] Add connectors to configunmarshaler test

### PR DESCRIPTION
**Pull request by djaglowski**
Subset of #6700 

This adds an unmarshaler to `otelcol/internal/configunmarshaler`.

### CODE REVIEW

**Comment by bogdandrutu** [comment]
Wait a bit to show you a better one.

**Comment by bogdandrutu** [comment]
Here is the alternative :)) https://github.com/open-telemetry/opentelemetry-collector/pull/6800

**Comment by djaglowski** [comment]
Rebased. Not much left to it

File: connector/connector_test.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ import (
 	"go.opentelemetry.io/collector/consumer"
 )
 
-func TestNewConnectorFactory_NoOptions(t *testing.T) {
+func TestNewFactory_NoOptions(t *testing.T) {
```
Comment (bogdandrutu) [inline connector/connector_test.go:27]: s/TestNewFactory_NoOptions/TestNewFactoryNoOptions

I was told that this is more "go" like :)

**Review by bogdandrutu — APPROVED** [review]

**Review by bogdandrutu — COMMENTED** [review]
I was expecting a change in https://github.com/open-telemetry/opentelemetry-collector/blob/main/otelcol/unmarshaler.go#L31 to add an entry for "connectors"

File: connector/connector_test.go (hunk 1 above)
Comment (djaglowski) [reply connector/connector_test.go]: Moved over here: https://github.com/open-telemetry/opentelemetry-collector/pull/6804

**Review by djaglowski — COMMENTED** [review]
> I was expecting a change in https://github.com/open-telemetry/opentelemetry-collector/blob/main/otelcol/unmarshaler.go#L31 to add an entry for "connectors"

Added, but we can't use it normally until `Factories` is updated to contain connector factories.

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**